netsblox.js

```diff
@@ -935,29 +935,19 @@ NetsBloxMorph.prototype.rawLoadCloudProject = function (project, isPublic) {
     this.source = 'cloud';
     project.Owner = project.Owner || SnapCloud.username;
     this.updateUrlQueryString(newRoom, isPublic === 'true');
-    if (project.SourceCode) {
-        var msg = this.showMessage('Opening project...');
-        return SnapActions.openProject(project.SourceCode)
-            .then(function() {
-                SnapCloud.projectId = projectId;
-                myself.room.silentSetRoomName(newRoom);
-                myself.room.ownerId = project.Owner;
-                myself.silentSetProjectName(roleName);
-
-                // Send the message to the server
-                myself.sockets.updateRoomInfo();
-                msg.destroy();
-            });
-    } else {  // initialize an empty code base
```

Comment:
was this a pure optimization else clause?

Reply:
It was actually a result of legacy code. Since the role might not have any code defined, it would just create a new role. However, since the addition of the openProject action, this can be handled by SnapActions